### PR TITLE
W-14749693-first draft

### PR DESCRIPTION
**Pull request by dmerlob**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 1]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
 | MIME Type
 |`*/java`
 |===
+
+
+[[java_support]]
+== Java Support
+
+Starting with version 2.6, DataWeave supports Java 17. Java 17 adds restrictions in encapsulation and reflective access that affects the xref:dataweave-formats-java.adoc[]. Because DataWeave uses Java's reflection API to read and write Java objects, ensure that the objects your application uses are Plain Old Java Objects (POJOs) containing:
```
Comment (mlischetti) [inline modules/ROOT/pages/dataweave-formats-java.adoc:310]: Set link label: `xref:dataweave-formats-java.adoc[Java Format]`

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 2]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+
+[source,java,linenums]
+----
+public class Item {
+	private String name;
+	private int quantity;
+	
+	public Item() {
+	}
+            public Item(String name, int quantity) {
```
Comment (mlischetti) [inline modules/ROOT/pages/dataweave-formats-java.adoc:383]: Remove extra _tab_

File: modules/ROOT/pages/dataweave-formats-java.adoc (hunk 1 above)
Comment (mlischetti) [inline modules/ROOT/pages/dataweave-formats-java.adoc:310]: Set link label: xref:dataweave-formats-java.adoc[Java Format]

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 3]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+
+[[java_support]]
+== Java Support
+
+Starting with version 2.6, DataWeave supports Java 17. Java 17 adds restrictions in encapsulation and reflective access that affects the xref:dataweave-formats-java.adoc[]. Because DataWeave uses Java's reflection API to read and write Java objects, ensure that the objects your application uses are Plain Old Java Objects (POJOs) containing:
+
+* Default constructor (required if your class needs to be instantiated)
+* Getters for all properties (required if your class needs to be read)
+* Setters for all properties (required if your class needs to be instantiated)
+
```
Comment (mlischetti) [inline modules/ROOT/pages/dataweave-formats-java.adoc:315]: I'm not sure that these clarifications ("required if your..") are needed.
I would like to keep it simple:
```
* Default constructor
* Getters for all properties
* Setters for all properties
```

**Review by mlischetti — APPROVED** [review]

File: modules/ROOT/pages/dataweave-formats-java.adoc (hunk 1 above)
Comment (mlischetti) [inline modules/ROOT/pages/dataweave-formats-java.adoc:310]: Set link label `xref:dataweave-formats-java.adoc[Java Format]`

File: modules/ROOT/pages/dataweave-formats-java.adoc (hunk 2 above)
Comment (mlischetti) [inline modules/ROOT/pages/dataweave-formats-java.adoc]: Remove extra _tab_

**Review by mlischetti — CHANGES_REQUESTED** [review]

**Review by mlischetti — APPROVED** [review]
LGTM

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 4]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+Starting with version 2.6, DataWeave supports Java 17. Java 17 adds restrictions in encapsulation and reflective access that affects the xref:dataweave-formats-java.adoc[]. Because DataWeave uses Java's reflection API to read and write Java objects, ensure that the objects your application uses are Plain Old Java Objects (POJOs) containing:
+
+* Default constructor
+* Getters for all properties
+* Setters for all properties
+
+[[java17_examples]]
+== Examples of Java Support
+
+The following examples run using Java 17 and using xref:dataweave-versioning-behavior.adoc[language level 2.6].
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
The following examples run using Java 17 and xref:dataweave-versioning-behavior.adoc[language level 2.6].
```

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 5]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+
+* <<example4>>
+* <<example5>>
+
+[[example4]]
+=== Example: Create a Java Object Instance
+
+This example shows how to create a Java object instance.
+
+==== User class
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
==== User Class
```

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 6]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+java.lang.IllegalAccessException - class 
+org.mule.weave.v2.module.pojo.reader.PropertyDefinition (in module org.mule.weave.module.java) cannot access a member of class User with modifiers "private
+----
+
+[[example5]]
+=== Example: Access the Values of Java Properties
+
+This example shows how to access the values of Java properties.
+
+==== Item class
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
==== Item Class
```

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 7]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+---
+{ 
+    "name": "payload.name",
+    "quantity": "payload.quantity"
+}
+----
+
+==== Output
+
+The expected output is a JSON.
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
The expected output is JSON.  
```

Comment (hannanelson) [reply modules/ROOT/pages/dataweave-formats-java.adoc]: Not sure if that's correct. Should it be "in JSON format" (or something like that)?

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 8]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+
+[source,java,linenums]
+----
+java.lang.IllegalAccessException: class 
+org.mule.weave.v2.module.pojo.reader.PropertyDefinition (in module org.mule.weave.module.java) cannot access a member of class Item with modifiers "private"
+----
+
+=== Solution
+
+To solve conflicts, ensure you:
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
To resolve conflicts, ensure that you: 
```

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 9]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+
+Because the item class definition doesn't have the proper getters, the output is `IllegalAccessException`.  
+
+[source,java,linenums]
+----
+java.lang.IllegalAccessException: class 
+org.mule.weave.v2.module.pojo.reader.PropertyDefinition (in module org.mule.weave.module.java) cannot access a member of class Item with modifiers "private"
+----
+
+=== Solution
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]: Is this the right term?

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 10]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+
+To solve conflicts, ensure you: 
+
+* Update your class definitions to follow the POJOs structure.
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
* Update your class definitions to follow the POJOs structure (recommended).
```

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 11]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+=== Solution
+
+To solve conflicts, ensure you: 
+
+* Update your class definitions to follow the POJOs structure. 
++
+We encourage you to adopt this option.
+* Use the compatibility flag `com.mulesoft.dw.java.disable_set_accessible=false` to allow accessible flag modification for reflection objects. 
++
+This approach is a workaround and works for customer domain classes only. Note that the usage of this compatibility flag may not affect future versions.
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
This approach is a workaround and works for customer domain classes only. Note that the usage of this compatibility flag might not affect future versions.
```

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 12]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+java.lang.IllegalAccessException: class 
+org.mule.weave.v2.module.pojo.reader.PropertyDefinition (in module org.mule.weave.module.java) cannot access a member of class Item with modifiers "private"
+----
+
+=== Solution
+
+To solve conflicts, ensure you: 
+
+* Update your class definitions to follow the POJOs structure. 
++
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion

```

File: modules/ROOT/pages/dataweave-formats-java.adoc [hunk 13]
```diff
@@ -302,3 +302,142 @@ This format supports the following MIME types.
...
+We encourage you to adopt this option.
```
Comment (hannanelson) [inline modules/ROOT/pages/dataweave-formats-java.adoc]:
```suggestion
```

Comment (hannanelson) [reply modules/ROOT/pages/dataweave-formats-java.adoc]: (Just an idea to streamline...)

File: modules/ROOT/pages/dataweave-formats-java.adoc (hunk 9 above)
Comment (dmerlob) [reply modules/ROOT/pages/dataweave-formats-java.adoc]: Maybe: 
- `Troubleshooting Common Errors` 
- `Conflict Management`

**Review by hannanelson — APPROVED** [review]

**Pull request merged**